scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,277 @@
+                    quest:setVar(player, 'prog', 3)
+                end,
+
+                [557] = function (player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 3 and
+                    npcUtil.popFromQM(player, npc, ID.mob.ALSHA, { claim = true, hide = 0 }) then
+                        return quest:messageSpecial(ID.text.FOUL_STENCH)
+                    end
+                end,
+
+                [558] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 5)
+                    player:addKeyItem(xi.ki.BETTER_HUMES_AND_MANNEQUINS)
+                    return quest:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.BETTER_HUMES_AND_MANNEQUINS)
+                end,
+            },
+        },
+
+        [xi.zone.MHAURA] =
+        {
+            ['Fyi_Chalmwoh'] =
+            {
+                onTrigger = function(player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 5 then
+                        return quest:progressEvent(321, { [0] = 704,
+                                                          [1] = xi.mannequin.getMannequins(player),
+                                                          [2] = xi.mannequin.cost.PURCHASE,
+                                                          [3] = xi.mannequin.cost.TRADE,
+                                                          [4] = xi.mannequin.cost.POSE,
+                                                          [5] = player:getGil(),
+                                                          [6] = xi.ki.BETTER_HUMES_AND_MANNEQUINS })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [321] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
+            },
+
+        }
+    },
+
+    -- Quest complete
+    {
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED and
+            player:hasKeyItem(xi.ki.BETTER_HUMES_AND_MANNEQUINS)
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Fyi_Chalmwoh'] =
+            {
+                onTrigger = function(player, csid, option, npc)
+                    return quest:event(321, { [1] = xi.mannequin.getMannequins(player),
+                                              [2] = xi.mannequin.cost.PURCHASE,
+                                              [3] = xi.mannequin.cost.TRADE,
+                                              [4] = xi.mannequin.cost.POSE,
+                                              [5] = player:getGil(),
+                                              })
+                end,
+
+                onTrade = function(player, npc, trade)
+                    -- Trade exactly one mannequin without gil.  Gil taken separately.
+                    if npcUtil.tradeHasExactly(trade, xi.items.HUME_M_MANNEQUIN) or
```

Comment:
multiline, also this should be a loop and then checked to prevent a gigantic block